flow/layers/physical_shape_layer.cc

```diff
@@ -75,6 +75,12 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
     // Let the system compositor draw all shadows for us, by popping us out as
     // a new frame.
     set_needs_system_composite(true);
+
+    // If the frame behind us is opaque, don't punch a hole in it for group
```

Comment:
This should only be required for Fuchsia ATM, because we want to do Opacity without allocating an offscreen buffer first.

The hole-punch trick allows to achieve correct group opacity for 1 layer of nesting, which is all Fuchsia needs at the moment.  The actual application of the opacity is delegated to the compositor, which applies it during the final pass before outputting to the display (one day we could even delegate this to the display hardware itself).

It's a bit of a hack at the moment, so I'd like to isolate the code for it as much as possible.  Once we move the SceneUpdate logic into platform-specific code that will be pretty easy

For now, I'll mark these with more OS_FUCHSIA.